src/codegen/object.rs

```diff
@@ -109,6 +111,96 @@ pub fn generate(
+                            };
+                        let (type_string, conversion) =
+                            match type_string.as_str() {
+                                "str" => ("&str", ".to_string()"),
+                                _ => (&*type_string, ""),
+                            };
+                        methods.push(format!("{}    pub fn {name}(mut self, {name}: {}) -> Self {{
+        self.{name} = Some({name}{});
+        self
+    }}", prefix, type_string, conversion, name=name));
+                        properties.push(property);
+                    },
+                    Err(_) => writeln!(w, "    //{}: /*Unknown type*/,", property.name)?,
+                }
+            }
+        }
+        writeln!(w, "}}\n")?;
+        writeln!(w, "#[cfg(any(feature = \"builders\", feature = \"dox\"))]")?;
+        writeln!(w, "impl {}Builder {{", analysis.name)?;
+        writeln!(w, "    pub fn new() -> Self {{")?;
+        writeln!(w, "        Self {{")?;
+        for property in &properties {
+            if let Some(version) = property.version {
+                writeln!(w, "            #[cfg(any(feature = \"{}\", feature = \"dox\"))]", version.to_feature())?;
+            }
+            let name = nameutil::mangle_keywords(nameutil::signal_to_snake(&property.name));
+            writeln!(w, "            {}: None,", name)?;
+        }
+        writeln!(w, "        }}")?;
+        writeln!(w, "    }}")?;
+        writeln!(w, "    pub fn build(self) -> {} {{", analysis.name)?;
+        writeln!(w, "        let mut n_properties = 0;")?;
+        writeln!(w, "        let mut property_names: Vec<CString> = vec![];")?;
+        writeln!(w, "        let mut names = vec![];")?;
+        writeln!(w, "        let mut values = vec![];")?;
+        for property in &properties {
+            let name = nameutil::mangle_keywords(nameutil::signal_to_snake(&property.name));
+            if let Some(version) = property.version {
+                writeln!(w, "        #[cfg(any(feature = \"{}\", feature = \"dox\"))]", version.to_feature())?;
+                writeln!(w, "        {{")?;
+            }
+            writeln!(w, "        if let Some({property}) = self.{property} {{", property=name)?;
+            writeln!(w, "            property_names.push(CString::new(\"{}\").unwrap());", property.name)?;
+            writeln!(w, "            names.push(property_names[property_names.len() - 1].as_ptr());")?;
+            writeln!(w, "            let property = {}.to_value();", name)?;
+            writeln!(w, "            values.push(property.into_raw());")?;
+            writeln!(w, "            n_properties += 1;")?;
+            writeln!(w, "        }}")?;
+            if property.version.is_some() {
+                writeln!(w, "        }}")?;
+            }
+        }
+        writeln!(w, "        unsafe {{")?;
+        writeln!(w, "            crate::Object::from_glib_none(gobject_sys::g_object_new_with_properties(")?;
```

Comment:
Why so complicated? :) `glib::Object::new()` can do exactly this here for you with only safe code